examples/beacon-api-sidecar-fetcher/src/main.rs

```diff
@@ -0,0 +1,228 @@
+                                    "should not fail to convert blob tx if it is already eip4844",
+                                ));
+                            }
+                        }
+                        Err(_err) => {
+                            // If a single BlobTransaction is missing we skip the queue.
+                            all_blobs_available = false;
+                        }
+                    };
+
+                    if all_blobs_available {
+                        this.queued_actions.extend(actions_to_queue);
+                    } else {
+                        let client_clone = this.client.clone();
+
+                        let sidecar_url = this.beacon_config.sidecar_url();
+
+                        // Query the Beacon Layer for missing BlobTransactions
+                        let query = Box::pin(async move {
+                            let response = match client_clone
+                                .get(sidecar_url)
+                                .header("Accept", "application/json")
+                                .send()
+                                .await
+                            {
+                                Ok(response) => response,
+                                Err(err) => return Err(SideCarError::ReqwestError(err)),
+                            };
+
+                            if !response.status().is_success() {
+                                return match response.status() {
+                                    StatusCode::BAD_REQUEST => Err(SideCarError::InvalidBlockID(
+                                        "Invalid request to server.".to_string(),
+                                    )),
+                                    StatusCode::NOT_FOUND => Err(SideCarError::BlockNotFound(
+                                        "Requested block not found.".to_string(),
+                                    )),
+                                    StatusCode::INTERNAL_SERVER_ERROR => {
+                                        Err(SideCarError::InternalError(
+                                            "Server encountered an error.".to_string(),
+                                        ))
+                                    }
+                                    _ => Err(SideCarError::UnknownError(
+                                        response.status().as_u16(),
+                                        "Unhandled HTTP status.".to_string(),
+                                    )),
+                                };
+                            }
+
+                            let bytes = match response.bytes().await {
+                                Ok(b) => b,
+                                Err(e) => return Err(SideCarError::NetworkError(e.to_string())),
+                            };
+
+                            let mut blobs_bundle: BlobsBundleV1 =
+                                match serde_json::from_slice(&bytes) {
```

Comment:
I don't think this is right, because the reponse object is slightly different, so we def need the types for this endpoint as well